### DISCUSSION TITLE
Editing array fields once at a time

### DISCUSSION THREAD

**Discussion by SafaAlfulaij** [react-hook-form/react-hook-form#8010] [General]
I'm trying to design a UI where users edit the array fields once at a time.
I'm not sure how to work properly with registering/unregistering controlled and uncontrolled inputs to show and save data correctly.

Example sandbox:
https://codesandbox.io/s/react-hook-form-usefieldarray-one-at-a-time-1pg5cu

Appreciate any help.

**Comment by bluebill1049** [comment]
If you want to do one at a time, you probably don't need useFieldArray, just control state with useState and calling `unregister` method for delete.

**Reply by SafaAlfulaij** [reply]
I think I got your point, but would you please "correct" the CSB when you have time? Thanks!

**Reply by SafaAlfulaij** [reply]
I got it working with hiding the other array fields.
Still, will appreciate any help.

**Reply by Moshyfawn** [reply]
Can u expand on your use-case a bit? What do you mean by "edit the array fields once at a time"? Is it the same form that you open with, say, different items IDs; like edit different user info?

**Reply by SafaAlfulaij** [reply]
Please check the CSB. I would like to display all the items in a dropdown, and the user would select one and edit it once-at-a-time.
As for now, there are several issues with showing and saving the data when you change the selection. This is what I'd like some clarity on.
I tried various combinations of `watch`, `register`, `unregister` with no proper final output, and the implementation feels weird.
